applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/memory.js

```diff
@@ -10,89 +10,63 @@ return baseclass.extend({
 	title: _('Memory'),
 
 	rrdargs: function(graph, host, plugin, plugin_instance, dtype) {
-		var p = [];
+		var definitions = [];
+		var instances;
 
-		var memory = {
-			title: "%H: Memory usage",
-			vlabel: "MB",
-			number_format: "%5.1lf%s",
-			y_min: "0",
-			alt_autoscale_max: true,
-			data: {
-				instances: {
-					memory: [
-						"free",
-						"buffered",
-						"cached",
-						"used"
-					]
-				},
+		function find_instances(dtype, wanted) {
+			var matching = graph.dataInstances(host, plugin, plugin_instance, dtype).filter(function(instance) {
+				return wanted.indexOf(instance) > -1;
+			});
 
-				options: {
-					memory_buffered: {
-						color: "0000ff",
-						title: "Buffered"
-					},
-					memory_cached: {
-						color: "ff00ff",
-						title: "Cached"
-					},
-					memory_used: {
-						color: "ff0000",
-						title: "Used"
-					},
-					memory_free: {
-						color: "00ff00",
-						title: "Free"
+			return matching.length ? { [dtype]: matching } : null;
```

Comment:
I think this needs to be changed to a list, and then it'll maintain the order.

Reply:
Here's a working solution (still [quadratic](https://accidentallyquadratic.tumblr.com/)).

```js
function find_instances(dtype, wanted) {
  var di = graph.dataInstances(host, plugin, plugin_instance, dtype);
  var matching = wanted.filter(w => di.includes(w));
  return matching.length ? { [dtype]: matching } : null;
}
````
Should be placed in both `cpu.js` and `memory.js`.

Reply:
@hippi-viking could you implement the fix?

Reply:
@raenye thanks for the solution proposal.
@systemcrash I'll have a go at it, but TBH I'm more of a backend-guy not really fluent in javascript. (And TBH it's good enough for my use as well the way it is, but when I'll have some time I'll definitely try to make it more aesthetically pleaseing.)

Reply:
(while you're at it, you should squash the commits and write a full name in the signoff line)